packages/components/app/styles/components/form/masked-input.scss

```diff
@@ -44,3 +48,9 @@ $hds-form-masked-input-button-size: 24px;
   border-color: transparent;
   cursor: pointer;
 }
+
+.hds-form-masked-input__copy-button {
+  grid-area: 1 / 2 / 2 / 3;
+  align-self: flex-start;
+  margin-left: 8px;
```

Comment:
Because the `VisibilityToggle` is always present we can't use the `grid-gap` property so we need to rely on `left-margin` on the `CopyButton` instead

Using `grid-gap: 8px`;
<img width="433" alt="Screenshot 2023-08-25 at 13 31 22" src="https://github.com/hashicorp/design-system/assets/788096/8f92d8b7-95dc-4c79-8db1-3608b3557c24">

Using `margin-left: 8px`;
<img width="431" alt="Screenshot 2023-08-25 at 13 31 03" src="https://github.com/hashicorp/design-system/assets/788096/85eeee0b-622e-48a3-b27e-21da59e32fd2">